### PR TITLE
CEDS-2495 - Add Dangerous Goods Code into C21 journey

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/controllers/declaration/IsExsController.scala [hunk 1]
```diff
@@ -66,12 +67,17 @@ class IsExsController @Inject()(
...
-      val updatedParties = answer.isExs match {
+      val updatedParties: Parties = answer.isExs match {
         case YesNoAnswers.yes => model.parties.copy(isExs = Some(answer))
         case YesNoAnswers.no  => model.parties.copy(isExs = Some(answer), carrierDetails = None, consignorDetails = None)
       }
 
-      model.copy(parties = updatedParties)
+      val updatedItems: Seq[ExportItem] = answer.isExs match {
+        case YesNoAnswers.yes => model.items.map(_.copy(dangerousGoodsCode = Some(UNDangerousGoodsCode(None))))
+        case YesNoAnswers.no  => model.items.map(_.copy(dangerousGoodsCode = None))
```
Comment (MaciejRewera) [inline app/controllers/declaration/IsExsController.scala:77]: 👍

File: app/controllers/navigation/Navigator.scala [hunk 2]
```diff
@@ -302,7 +302,8 @@ object Navigator {
   }
 
   val clearanceCacheItemDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode, String) => Call] = {
-    case CommodityMeasure => commodityMeasureClearancePreviousPage
+    case CommodityMeasure   => commodityMeasureClearancePreviousPage
+    case PackageInformation => packageInformationClearancePreviousPage
```
Comment (MaciejRewera) [inline app/controllers/navigation/Navigator.scala:306]: I feel like the case for `PackageInformation` should be removed from `clearanceItemPage`, isn't it?

Comment (SteveSugden) [reply app/controllers/navigation/Navigator.scala:306]: You are correct Maciej

File: test/unit/controllers/declaration/CommodityDetailsControllerSpec.scala [hunk 3]
```diff
@@ -138,15 +139,48 @@ class CommodityDetailsControllerSpec extends ControllerSpec with OptionValues {
     }
 
     onJourney(CLEARANCE) { request =>
-      "return 303 (SEE_OTHER) and redirect to UN Dangerous Goods Code page" in {
-
-        withNewCaching(request.cacheModel)
+      def controllerRedirectsToNextPageForProcedureCodeAndExsStatus(procedureCode: String, exsStatus: String, call: Call) = {
```
Comment (MaciejRewera) [inline test/unit/controllers/declaration/CommodityDetailsControllerSpec.scala:142]: Nit: Could you change the name to `expectedCall` (I assume this is this field's purpose).

File: test/unit/controllers/declaration/UNDangerousGoodsCodeControllerSpec.scala [hunk 4]
```diff
@@ -111,11 +111,24 @@ class UNDangerousGoodsCodeControllerSpec extends ControllerSpec with OptionValue
...
+
+          await(result) mustBe aRedirectToTheNextPage
+          thePageNavigatedTo mustBe controllers.declaration.routes.CusCodeController.displayPage(Mode.Normal, itemId)
+          verify(mockPage, times(0)).apply(any(), any(), any())(any(), any())
+        }
+
+      }
+
+      onJourney(CLEARANCE) { request =>
+        def controllerRedirectsToNextPageForProcedureCode(procedureCode: String, call: Call) = {
```
Comment (MaciejRewera) [inline test/unit/controllers/declaration/UNDangerousGoodsCodeControllerSpec.scala:129]: Nit: Same here with renaming to `expectedCall`

**Pull request merged**